R/robustbayesianmetaanalysis.R

```diff
@@ -1756,248 +1754,248 @@ RobustBayesianMetaAnalysis <- function(jaspResults, dataset, options, state = NU
   fit   <- jaspResults[["model"]][["object"]]
   
   # select models to iterate over
-  if (options[["diagnostics_single"]]) {
-    models_i <- options[["diagnostics_single_model"]]
-    if (models_i < 1 || models_i > length(fit[["models"]])) {
-      temp_model  <- createJaspContainer(title = gettextf("Model %i", models_i))
-      diagnostics[[paste0("model_", models_i)]] <- temp_model
-      temp_error  <- createJaspPlot(title = "")
-      temp_error$dependOn("diagnostics_single_model", "diagnostics_single")
-      temp_error$setError(gettextf("Model %1$i does not exist. Select one of the models between 1 and %2$i.", models_i, length(fit[["models"]])))
-      temp_model[["temp_error"]] <- temp_error
+  if (options[["diagnosticsSingle"]]) {
+    modelsI <- options[["diagnosticsSingleModel"]]
+    if (modelsI < 1 || modelsI > length(fit[["models"]])) {
+      tempModel  <- createJaspContainer(title = gettextf("Model %i", modelsI))
+      diagnostics[[paste0("model_", modelsI)]] <- tempModel
+      tempError  <- createJaspPlot(title = "")
+      tempError$dependOn("diagnosticsSingleModel", "diagnosticsSingle")
+      tempError$setError(gettextf("Model %1$i does not exist. Select one of the models between 1 and %2$i.", modelsI, length(fit[["models"]])))
+      tempModel[["tempError"]] <- tempError
       return()
     }
   } else{
-    models_i <- 1:length(fit[["models"]])
+    modelsI <- 1:length(fit[["models"]])
   }
   
   # collect the parameters
-  parameters <- c(if (options[["diagnostics_mu"]])
+  parameters <- c(if (options[["diagnosticsMu"]])
     "mu",
-    if (options[["diagnostics_tau"]])
+    if (options[["diagnosticsTau"]])
       "tau",
-    if (options[["diagnostics_omega"]])
+    if (options[["diagnosticsOmega"]])
       "omega",
-    if (options[["diagnostics_theta"]])
+    if (options[["diagnosticsTheta"]])
       "theta")
   
   # do the iteration
-  for (i in models_i) {
+  for (i in modelsI) {
```

Comment:
don't you want to some how just sequence along?

Reply:
no, because it can refer to an index of specific model

```
  if (options[["resultsIndividualSingle"]]) {
    modelsI <- options[["resultsIndividualSingleNumber"]]
```